currency_rate_update/models/res_currency_rate.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2019 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields, api
+
+
+class ResCurrencyRate(models.Model):
+    _name = 'res.currency.rate'
+    _inherit = ['res.currency.rate', 'mail.thread']
```

Comment:
I think tracking the rate changes is not adding too much, but overheading everything, so I would remove this inheritance and remove the `track_visibility`.

Reply:
In most cases, currency rate is set once, and changing it for a specific date is a special event that should be logged, which would be useful to understand why values have changed. As far as I understand, initially it will not generate any emails, just change records.

Reply:
But with this, the cron will also log an entry each day.

Reply:
for each _new_ day, to be precise

Reply:
Old days would be kept as is unless manually updated or requested to update.

Reply:
I still think it's not needed. It doesn't add too much value. You already have the rates with a date reference...

Reply:
Yet if it was used and then changed - it's better to keep at least some record of rate being altered

Reply:
But overhead each record with this inheritance?

Reply:
How much this overhead would be?

Reply:
Each time the record is accessed, a lot of checks are done for example. Check mail.thread code for getting you a bit scared.

Reply:
I gave it some thought - seems still valid to me

Reply:
Otherwise it would be super-hard to track down the changes